source/unified-test-format/unified-test-format.rst

```diff
@@ -929,6 +935,21 @@ The structure of this object is as follows:
   to ``cmap`` for both would either be redundant (if the ``events`` arrays were
   identical) or likely to fail (if the ``events`` arrays differed).
 
+
+.. _test_expectLogMessages:
+
+- ``expectLogMessages``: Optional array of one or more `expectedLogMessagesForClient`_
+  objects. For one or mroe clients, a list of log messages that are expected to
+  be observed in a particular order.
+
+  If a driver only supports configuring log collectors globally (for all
```

Comment:
Reviewing the Java driver POC, I recall a conversation about using `uriOptions,appName` as a sneaky way for the Java driver to get around the problem of a global log collector. There is even an assertion that requires it:

```
        this.applicationName = requireNonNull(applicationName);
```

I can't recall the outcome of our discussion, but I don't see a way to implement this in Java without using appname as a way to control which messages are intercepted.  Can we consider making it a requirement to specify appname whenever log messages are being observed?

Reply:
That could certainly be done, yes.  The only potential issue is the entity names are typically short and not unique, but then so might appName.  And in reality we should be closing all of our MongoClient instances in our tests so at most there should be just a few that are active at a time: the one currently under test, and the "global" one that is used for all the ad hoc integration tests.